card_house/controller.py

```diff
@@ -0,0 +1,73 @@
+import os
+from typing import List, Dict, Any
+
+from card_house.cards_exporter.cards_exporter.icards_exporter import ICardsExporter
+from card_house.db.error import CardExists
+from card_house.infrastructure.cards_creators.db_cards_creator import DBCardsCreator
+from card_house.infrastructure.cards_creators.json_to_cards_creator import JsonToCardsCreator
+from card_house.infrastructure.entities.db_card import DBCard, CardKeys
+from card_house.db.iclient_db import IClientDB
+
+from flask import Flask, request
+
+
+CARDS_SERIALIZED = Dict[str, List[Dict[str, str]]]
+
+
+class Controller:
+    def __init__(
+            self,
+            download_path: str,
+            client_db: IClientDB,
+            db_cards_creator: DBCardsCreator,
+            cards_exporter: ICardsExporter,
+            cards_creator: JsonToCardsCreator,
+            app: Flask
+    ):
+        self.download_path = download_path
+        self.app = app
+        self.cards_exporter = cards_exporter
+        self.db_card_creator = db_cards_creator
+        self.cards_creator = cards_creator
+        self.client_db = client_db
+        self._setup_routes()
+
+    def _setup_routes(self):
+        self.app.route("/add_card", methods=["POST", "OPTIONS"])(self.add_card)
+        self.app.route("/remove_card", methods=["DELETE"])(self.remove_card)
+        self.app.route("/export_cards", methods=["POST"])(self.export_cards)
+        self.app.route("/filter_cards", methods=["POST"])(self.filter_cards)
+
+    def run(self):
+        self.app.run(debug=True)
+
+    def add_card(self) -> str:
+        card = self.cards_creator.create_card(request)
+        db_card = self.db_card_creator.create_card(card)
+        card_with_same_properties = self.client_db.get_cards(
+            card_properties={
+                CardKeys.TYPE: db_card.type.value,
+                CardKeys.CONTENT: db_card.content,
+            }
+        )
+
+        if len(card_with_same_properties) != 0:
+            raise CardExists
+
+        card_id = self.client_db.add_card(db_card)
+        return card_id
+
+    def remove_card(self) -> str:
```

Comment:
Good with the flow of this method, it's clear, readable and describes exactly the flow of the action and not the logic inside. That's how I see methods of controller.